### PR TITLE
Add option to restrict allowed data fields

### PR DESCRIPTION
**Pull request by maxsmythe**
Signed-off-by: Max Smythe <smythe@google.com>

### CODE REVIEW

File: constraint/pkg/client/client.go [hunk 1]
```diff
@@ -95,6 +95,20 @@ func Targets(ts ...TargetHandler) ClientOpt {
...
 }
 
+// AllowedDataFields sets the fields under `data` that Rego in ConstraintTemplates
+// can access. If unset, all fields can be accessed. Only fields recognized by
+// the system can be enabled.
+func AllowedDataFields(fields ...string) ClientOpt {
+	return func(c *client) error {
+		var errs Errors
+		c.allowedDataFields = fields
+		if len(errs) > 0 {
```
Comment (ritazh) [inline constraint/pkg/client/client.go]: where does this `errs` get updated?

File: constraint/pkg/client/client_test.go [hunk 2]
```diff
@@ -606,3 +606,66 @@ func TestRemoveConstraint(t *testing.T) {
...
+			Handler:       &badHandler{Name: "h1", HasLib: true},
+			Template:      createTemplate(name("fakes"), crdNames("Fakes"), targets("h1")),
+			ErrorExpected: false,
+		},
+		{
+			Name:          "Inventory Used",
+			Handler:       &badHandler{Name: "h1", HasLib: true},
+			Template:      inventoryTempl,
+			ErrorExpected: true,
+		},
```
Comment (ritazh) [inline constraint/pkg/client/client_test.go]: Should we add a test case where `Inventory` used is allowed?

File: constraint/pkg/client/client.go (hunk 1 above)
Comment (maxsmythe) [reply constraint/pkg/client/client.go]: oops, this should go

File: constraint/pkg/client/client_test.go (hunk 2 above)
Comment (maxsmythe) [reply constraint/pkg/client/client_test.go]: Good idea, added.

File: constraint/pkg/client/backend.go [hunk 3]
```diff
@@ -54,6 +59,7 @@ func (b *Backend) NewClient(opts ...ClientOpt) (Client, error) {
 	if len(errs) > 0 {
 		return nil, errs
```
Comment (ritazh) [inline constraint/pkg/client/backend.go]: Isn't this returning an array but expecting an `error`?

File: constraint/pkg/client/client_test.go [hunk 4]
```diff
@@ -606,3 +606,78 @@ func TestRemoveConstraint(t *testing.T) {
...
+	for _, tt := range tc {
+		t.Run(tt.Name, func(t *testing.T) {
+			d := local.New()
+			b, err := NewBackend(Driver(d))
+			if err != nil {
+				t.Fatalf("Could not create backend: %s", err)
+			}
+			f := AllowedDataFields()
+			if tt.InvAllowed {
+				f = AllowedDataFields("inventory")
```
Comment (ritazh) [inline constraint/pkg/client/client_test.go]: can we add another test for if `f = AllowedDataFields("notallowed")`? and make sure NewClient returns an empty c.allowedDataFields

File: constraint/pkg/client/backend.go (hunk 3 above)
Comment (maxsmythe) [reply constraint/pkg/client/backend.go]: The array obeys the Error interface

File: constraint/pkg/client/client_test.go (hunk 4 above)
Comment (maxsmythe) [reply constraint/pkg/client/client_test.go]: done
